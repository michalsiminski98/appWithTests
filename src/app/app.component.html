<app-add-form></app-add-form>
<app-hide-button>SHOW / HIDE</app-hide-button>
<app-best-quote [title]= "titleForBest"></app-best-quote>
<app-quote-list [title] = "titleForList"></app-quote-list>
<app-worst-list [title] = "titleForWorst"></app-worst-list>