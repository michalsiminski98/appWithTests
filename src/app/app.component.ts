import { Component } from '@angular/core';

@Component({
  selector: 'app-root',
  templateUrl: './app.component.html',
  styleUrls: ['./app.component.css']
})
export class AppComponent {
  titleForBest: string = "Best Quote:"
  titleForList: string = "Quotes list:";
  titleForWorst: string = "Worst Quotes:";

  public isSubmitting: boolean = false;

  showForm = false;

  onShowForm(): void{
    this.showForm = !this.showForm;
  }

  adding(a:number, b:number): number{
    return a + b
  }
}

