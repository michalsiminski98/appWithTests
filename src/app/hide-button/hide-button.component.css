button{
  width: 110px;
  height: 40px;
  padding: 0 8px;
  line-height: 40px;
  margin-left: 50vw;
  transform: translate(-50%, 16px);
  border: 2px solid black;
  background-color: white;
  color: black;
  cursor: pointer;
  transition: .2s;
}
button:hover{
  color: white;
  border-color: white;
  background-color: black;
}