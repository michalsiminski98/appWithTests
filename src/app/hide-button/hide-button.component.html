<p>hide-button works!</p>
