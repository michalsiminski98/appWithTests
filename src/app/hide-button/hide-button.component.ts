import { Component, EventEmitter, Output } from '@angular/core';

@Component({
  selector: 'app-hide-button',
  templateUrl: './hide-button.component.html',
  styleUrls: ['./hide-button.component.css']
})
export class HideButtonComponent {

  @Output() showForm = new EventEmitter<void>();

  constructor() { }
}
