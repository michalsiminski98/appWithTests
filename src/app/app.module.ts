import { NgModule } from '@angular/core';
import { BrowserModule } from '@angular/platform-browser';

import { AppComponent } from './app.component';
import { HideButtonComponent } from './hide-button/hide-button.component';
import { AddFormComponent } from './add-form/add-form.component';
import { QuoteListComponent } from './quote-list/quote-list.component';
import { BestQuoteComponent } from './best-quote/best-quote.component';
import { WorstListComponent } from './worst-list/worst-list.component';
import { FormsModule } from '@angular/forms';

@NgModule({
  declarations: [
    AppComponent,
    HideButtonComponent,
    AddFormComponent,
    QuoteListComponent,
    BestQuoteComponent,
    WorstListComponent
  ],
  imports: [
    BrowserModule,
    FormsModule
  ],
  providers: [],
  bootstrap: [AppComponent]
})
export class AppModule { }
