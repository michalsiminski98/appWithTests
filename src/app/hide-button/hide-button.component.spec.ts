import { Shallow } from "shallow-render";
import { AppModule } from "../app.module";
import { HideButtonComponent } from "./hide-button.component";

describe('HideButtonComponent', () => {
  let shallow: Shallow<HideButtonComponent>;

  beforeEach(() => {
    shallow = new Shallow(
      HideButtonComponent,
      AppModule
    );
  });

  it('should match snapshot', async () => {
    const { instance } = await shallow.render();
    expect(instance).toMatchSnapshot();
  })
});
