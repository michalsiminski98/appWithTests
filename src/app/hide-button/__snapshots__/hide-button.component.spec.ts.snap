// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HideButtonComponent should match snapshot 1`] = `
HideButtonComponent {
  "showForm": EventEmitter_ {
    "__isAsync": false,
    "_isScalar": false,
    "closed": false,
    "emit": [MockFunction],
    "hasError": false,
    "isStopped": false,
    "observers": Array [],
    "thrownError": null,
  },
}
`;
